proxy/http/remap/NextHopSelectionStrategy.cc

```diff
@@ -214,8 +221,13 @@ NextHopSelectionStrategy::Init(ts::Yaml::Map &n)
               std::shared_ptr<HostRecord> host_rec = std::make_shared<HostRecord>(hosts_list[hst].as<HostRecord>());
               host_rec->group_index                = grp;
               host_rec->host_index                 = hst;
-              if (mach->is_self(host_rec->hostname)) {
+              if ((self_host == host_rec->hostname) || mach->is_self(host_rec->hostname.c_str())) {
+                if (grp != 0) {
+                  throw std::invalid_argument("self host (" + self_host + ") can only appear in first host group");
+                }
                 h_stat.setHostStatus(host_rec->hostname.c_str(), TSHostStatus::TS_HOST_STATUS_DOWN, 0, Reason::SELF_DETECT);
```

Comment:
@ywkaras the self detection doesn't seem to work now.  Host status no-longer has a SELF_DETECT down for the self_host.  I'm using the peering ring mode.